test/static.js

```diff
@@ -304,3 +381,242 @@ test('/documentation/:file should be served from custom location with trailing s
+  const config = {
+    specification: {
+      path: './examples/example-static-specification.yaml'
+    },
+    exposeRoute: true
+  }
+
+  t.plan(8)
+  const fastify = Fastify()
+  fastify.register(fastifySwagger, config)
+
+  fastify.inject(
+    {
+      method: 'GET',
+      url: '/documentation/yaml'
+    },
+    (err, res) => {
+      t.error(err)
+      t.is(typeof res.payload, 'string')
+      t.is(res.headers['content-type'], 'application/x-yaml')
+      try {
+        yaml.safeLoad(res.payload)
+        t.pass('valid swagger yaml')
+      } catch (err) {
+        t.fail(err)
+      }
+    }
+  )
+
+  fastify.inject(
+    {
+      method: 'GET',
+      url: '/documentation/yaml'
+    },
+    (err, res) => {
+      t.error(err)
+      t.is(typeof res.payload, 'string')
+      t.is(res.headers['content-type'], 'application/x-yaml')
+      try {
+        yaml.safeLoad(res.payload)
+        t.pass('valid swagger yaml')
+      } catch (err) {
+        t.fail(err)
+      }
+    }
+  )
+})
+
+test('/documentation/json returns cache.swaggerObject on second request in dynamic mode', t => {
+  const config = {
+    specification: {
+      path: './examples/example-static-specification.json'
+    },
+    exposeRoute: true
+  }
+
+  t.plan(8)
+  const fastify = Fastify()
+  fastify.register(fastifySwagger, config)
+
+  fastify.inject(
+    {
+      method: 'GET',
+      url: '/documentation/json'
+    },
+    (err, res) => {
+      t.error(err)
+      t.is(typeof res.payload, 'string')
+      t.is(res.headers['content-type'], 'application/json; charset=utf-8')
+      try {
+        t.pass('valid swagger json')
+      } catch (err) {
+        t.fail(err)
+      }
+    }
+  )
+
+  fastify.inject(
+    {
+      method: 'GET',
+      url: '/documentation/json'
+    },
+    (err, res) => {
+      t.error(err)
+      t.is(typeof res.payload, 'string')
+      t.is(res.headers['content-type'], 'application/json; charset=utf-8')
+      try {
+        t.pass('valid swagger json')
+      } catch (err) {
+        t.fail(err)
+      }
+    }
+  )
+})
+
+test('prints to console when swagger routes are not exposed', t => {
+  const config = {
+    mode: 'static',
+    specification: {
+      path: './examples/example-static-specification.json'
+    },
+    exposeRoute: false
+  }
+
+  t.plan(4)
+  const fastify = Fastify()
+  fastify.register(fastifySwagger, config)
+
+  // check that yaml is there
+  fastify.inject(
+    {
+      method: 'GET',
+      url: '/documentation/json'
+    },
+    (err, res) => {
+      t.error(err)
+      t.is(typeof res.payload, 'string')
+      t.is(res.headers['content-type'], 'application/json; charset=utf-8')
+      try {
+        t.pass('routes are not exposed')
+      } catch (err) {
+        t.fail(err)
+      }
+    }
+  )
+})
+
+test('inserts default opts in fastifySwagger', t => {
+  t.plan(1)
+  const fastify = Fastify()
+  const next = () => console.log()
```

Comment:
why?

Reply:
I needed to pass `next` function as a param to `fastifySwagger`, and the `next` returns `void`. I didn't find a better solution to this.

Reply:
_never_ use `console.log`. You can add an assertion there with `t.error(err)` to assert that `next` is called without errors.